inst/tinytest/test_d_delayed_7.R

```diff
@@ -0,0 +1,167 @@
+#
+# In case this needs to be recreated in TileDB Cloud, here's how:
+#
+# myfunc <- function(df, attr, exponent) {
+#   vec <- as.vector(df[[attr]])
+#   sum(vec ** exponent)
+# }
+# register_udf(namespace='johnkerl-tiledb', name='tiledb-cloud-r-single-array-udf-r', type='single_array', func=myfunc)
+
+# Non-delayed
+o <- execute_array_udf(
+  namespace=namespaceToCharge,
+  array="TileDB-Inc/quickstart_dense",
+  registered_udf_name='johnkerl-tiledb/tiledb-cloud-r-single-array-udf-r',
+  selectedRanges=list(cbind(1,4), cbind(1,4)),
+  attrs=c("a"),
+  args=list(attr="a", exponent=3)
+)
+expect_equal(o, 18496)
+
+# Delayed
+a <- delayed_array_udf(
+  namespace=namespaceToCharge,
+  array="TileDB-Inc/quickstart_dense",
+  registered_udf_name='johnkerl-tiledb/tiledb-cloud-r-single-array-udf-r',
+  selectedRanges=list(cbind(1,4), cbind(1,4)),
+  attrs=c("a"),
+  args=list(attr="a", exponent=3),
+  name='my single-array udf'
+)
+o <- compute(a)
+expect_equal(o, 18496)
+
+# ----------------------------------------------------------------
+# Delayed registered Python generic UDF
+
+# This one we require to be already registered in prod -- we don't dynamically
+# register it, then test it, then deregister it -- since it's a Python UDF.
+#
+# In case this needs to be recreated in TileDB Cloud, here's how:
+#
+# import tiledb, tiledb.cloud
+# def myfunc(vec, exponent):
+#   import numpy
+#   # Cast from NumPy type to Python type so this can serialize via JSON/Arrow to R callers
+#   return int(numpy.sum(numpy.array(vec) ** exponent))
+# tiledb.cloud.udf.register_generic_udf(myfunc, name="tiledb-cloud-r-generic-udf-py", namespace="johnkerl-tiledb")
+
+# Non-delayed
+o <- execute_generic_udf(
+  namespace=namespaceToCharge,
+  registered_udf_name='johnkerl-tiledb/tiledb-cloud-r-generic-udf-py',
+  args=list(vec=1:16, exponent=3),
+  language='python'
+)
+expect_equal(o, 18496)
+
+# Delayed
+a <- delayed_generic_udf(
+  namespace=namespaceToCharge,
+  registered_udf_name='johnkerl-tiledb/tiledb-cloud-r-generic-udf-py',
+  args=list(vec=1:16, exponent=3),
+  name='my generic udf',
+  language='python'
+)
+o <- compute(a)
+expect_equal(o, 18496)
+
+# ----------------------------------------------------------------
+# Delayed registered Python single-array UDF
+
+# In case this needs to be recreated in TileDB Cloud, here's how:
+#
+# import tiledb, tiledb.cloud
+# # 'df' is an OrderedDict from attribute name to multi-dimensional numpy.array.
+# # OrderedDict([('a', array([[ 1,  2,  3,  4], [ 5,  6,  7,  8], [ 9, 10, 11, 12], [13, 14, 15, 16]], dtype=int32))])
+# def myfunc(df, attr, exponent):
+#   import numpy
+#   # Cast from NumPy type to Python type so this can serialize via JSON/Arrow to R callers
+#   return int(numpy.sum(df[attr] ** exponent))
+# tiledb.cloud.udf.register_single_array_udf(myfunc, name="tiledb-cloud-r-single-array-udf-py", namespace="johnkerl-tiledb")
+
+# PENDING https://github.com/TileDB-Inc/TileDB-REST-UDF-Docker-Images/pull/213
```

Comment:
I'll defer vignetting either generic or array UDFs until this PR is merged and deployed into prod, as calling registered Python array UDFs does not work without this PR.